src/main/java/stelnet/ui/Renderable.java

```diff
@@ -28,4 +45,29 @@ public void log() {
     public void log(Position position) {
         LogHelper.debug(String.format("Rendered %s in %s", this, position));
     }
+
+    public Size getSize() {
+        if (size == null) {
+            LogHelper.warn("Null size found");
+        }
+        return size;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public Position getOffset() {
+        return offset;
+    }
+
+    public void setLocation(Location location) {
+        this.location = location;
+    }
+
+    public void setOffset(Position offset) {
+        this.offset = offset;
+    }
```

Comment:
JFYI there is a library called project lombok, which during compile time creates getters and setters.

So you can add an annotation @Data to a class and it will generate you all setters/getters for class instance variables. And you can still create your own as well.

https://github.com/mnukka/spring-security-base/blob/master/src/main/java/ee/company/crm/domain/persistence/user/UserEntity.java
vs
https://github.com/mnukka/spring-security-base/blob/Helmes/src/main/java/ee/company/crm/domain/persistence/user/UserEntity.java

More info how to set it up if you should feel like you want to incorporate it:
https://www.baeldung.com/lombok-ide

Reply:
Don't use either. Well, kinda. Vim + jdt.ls (so what Eclipse use for LSP). Then just Gradle. I'll add it if I can make it work in Gradle and project's Java (7) is not an issue.